libcontainer/configs/validate/validator.go

```diff
@@ -16,30 +16,19 @@ import (
 	"golang.org/x/sys/unix"
 )
 
-type Validator interface {
-	Validate(*configs.Config) error
-}
-
-func New() Validator {
-	return &ConfigValidator{}
-}
-
-type ConfigValidator struct{}
-
 type check func(config *configs.Config) error
 
-func (v *ConfigValidator) Validate(config *configs.Config) error {
+func Validate(config *configs.Config) error {
```

Comment:
Thanks for doing this! I recall looking at this a couple of times (why the "Validator"??)

Perhaps as a follow-up, we should move `configs/validate` to `configs`. It's a bit weird to have it in a separate package, as it's validating the config. Moving it into configs, if could just be `validate(config)` (no need to export), or

```go
func (config *configs.Config) validate() error
```

Reply:
> Perhaps as a follow-up, we should move `configs/validate` to `configs`.

I gave it a try and got circular deps as a result. Will take a look later, but it seems this was the reason why it's in a separate package.

Oh, and we still have to export it because it's also used from libcontainer/specconv/spec_linux_test.go